econplayground/main/views.py

```diff
@@ -221,9 +223,23 @@ class GraphEmbedPublicMinimalView(
     template_name = 'main/graph_embed_public_minimal.html'
 
 
-class GraphDeleteView(UserPassesTestMixin, DeleteView):
+class GraphDeleteView(LoginRequiredMixin, CohortInstructorMixin, DeleteView):
     model = Graph
-    success_url = '/'
+
+    def get_context_data(self, **kwargs):
+        ctx = super(GraphDeleteView, self).get_context_data(**kwargs)
+        ctx.update({
+            'cohort': self.cohort,
+        })
+        return ctx
+
+    def get_success_url(self):
+        messages.add_message(
+            self.request, messages.SUCCESS,
+            '<strong>{}</strong> has been deleted.'.format(self.object.name),
+            extra_tags='safe')
+
+        return reverse('topic_list', kwargs={'cohort_pk': self.cohort.pk})
 
     def test_func(self):
```

Comment:
If you're switching from UserPassesTestMixin to CohortInstructorMixin, you can remove this test_func here, since it won't be used.